contracts/CourtRoom.sol

```diff
@@ -7,24 +7,32 @@ import "./ensabstract.sol";
 
 contract CaseContract is Owned {
 
+  struct promise{
+		address beneficiary;
+		uint256 blockNumber;
+		uint8 sig_v;
+		bytes32 sig_r;
+		bytes32 sig_s;
+	}
 	struct claim {
 		bytes32 claimId;
 		address plaintiff;
 		bytes32[] evidence;
+		promise   servicePromise;
 		uint status;
 		bool valid;
 	}
 
 	//claimid map to claim
-  mapping(bytes32 => claim) public OpenClaims;
+  mapping(bytes32 => claim)  OpenClaims;
 
 
 	function CaseContract() {
```

Comment:
removed "public" due to  Error: Internal type is not allowed for public state variables.
The alternative is to flatten the struct ...but it is not nice.